blockly/src/utils/pathfinding/BFSPathFinding.java

```diff
@@ -0,0 +1,121 @@
+package utils.pathfinding;
+
+import core.level.utils.Coordinate;
+import java.util.*;
+import utils.LevelUtils;
+
+/**
+ * Breadth-First Search (BFS) pathfinding algorithm implementation.
+ *
+ * <p>This class implements the IPathfindingLogic interface and provides the logic for performing
+ * BFS pathfinding. It maintains the open and closed sets, tracks the path, and provides methods to
+ * initialize the search, perform steps, and retrieve the final path.
+ *
+ * @see IPathfindingLogic
+ * @see systems.PathfindingSystem PathfindingSystem
+ */
+public class BFSPathFinding implements IPathfindingLogic {
+  private Queue<Coordinate> queue;
+  private Set<Coordinate> openSet;
+  private Set<Coordinate> closedSet;
+  private Map<Coordinate, Coordinate> cameFrom;
+  private Coordinate start;
+  private Coordinate end;
+  private Coordinate lastProcessed;
+  private boolean isFinished;
+
+  @Override
+  public void initialize(Coordinate start, Coordinate end) {
+    this.queue = new LinkedList<>();
+    this.openSet = new HashSet<>();
+    this.closedSet = new HashSet<>();
+    this.cameFrom = new HashMap<>();
+    this.start = start;
+    this.end = end;
+    this.lastProcessed = null;
+    this.isFinished = false;
+
+    // Add starting node to the queue and open set
+    queue.add(start);
+    openSet.add(start);
+  }
+
+  @Override
+  public void performStep() {
+    if (queue.isEmpty() || isFinished) {
+      isFinished = true;
+      return;
+    }
+
+    // Get the next node from the queue
+    Coordinate current = queue.poll();
+    lastProcessed = current;
+
+    // Move current from an open set to closed set
+    openSet.remove(current);
+    closedSet.add(current);
+
+    // Check if we've reached the end
+    if (current.equals(end)) {
+      isFinished = true;
+      return;
+    }
+
+    // Get valid neighbors (4-way movement)
+    List<Coordinate> neighbors =
+        LevelUtils.walkableNeighbors(current).stream()
+            .filter(neighbor -> !closedSet.contains(neighbor))
+            .toList();
+
+    // Process each valid neighbor
+    for (Coordinate neighbor : neighbors) {
+      if (!openSet.contains(neighbor)) {
+        // Add to queue and open set
+        queue.add(neighbor);
+        openSet.add(neighbor);
```

Comment:
Was ist das Ziel der `openSet`?

Die Standard-Breitensuche arbeitet einfach nur mit einer Queue. Du willst scheinbar die Graphsearch-Variante haben mit der extra Queue für markierte Knoten (`closedSet`). Aber mir ist der Sinn der `openSet`-Queue nicht klar? Duplikate in der Arbeitsqueue kann es eh nicht geben, weil die Queue ja die Mengeneigenschaft hat?

Reply:
Ich verwende `closedSet` und `openSet`  für die Darstellung die Path-Findings suchen. Ich fande es sinnvoll diese Variante zu zeigen, aber ansonsten ist das noch ein offenes Thema für das Meeting mit Benjamin und den SHKs